umbral/pre.py

```diff
@@ -108,6 +115,83 @@ def from_bytes(cls, capsule_bytes: bytes, curve: ec.EllipticCurve = None):
         components = splitter(capsule_bytes)
         return cls(*components)
 
+    def get_or_set_delegating_key(self, delegating_key: UmbralPublicKey = None):
```

Comment:
Hm.. I wonder if it can be done through `__getattr__` / `__setattr__`. Of course, that'd be too low-level (and, hence, error-prone). So, I wonder if Python has something to construct a "set-once" property conveniently

Reply:
Like, using a setter:
```
In [1]: class X(object):
   ...:     def __init__(self):
   ...:         self.__x = None
   ...:     @property
   ...:     def x(self):
   ...:         return self.__x
   ...:     @x.setter
   ...:     def x(self, value):
   ...:         if self.__x is not None:
   ...:             raise ValueError('oops')
   ...:         else:
   ...:             self.__x = value
   ...:             

In [2]: y = X()

In [3]: y.x

In [4]: y.x = 1

In [5]: y.x
Out[5]: 1

In [6]: y.x = 1
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-6-183902452f08> in <module>()
----> 1 y.x = 1

<ipython-input-1-118622fe2a80> in x(self, value)
      8     def x(self, value):
      9         if self.__x is not None:
---> 10             raise ValueError('oops')
     11         else:
     12             self.__x = value

ValueError: oops
```

Reply:
Yeah, I thought about it - I can do that instead if you prefer.

Reply:
I think, that'd be more readable where it's used.
Yeah, also what feel strange also is that we repeat _exactly the same thing_ multiple times

Reply:
Heh, yeah.  When I first wrote it, I was thinking it was only going to be once.  Then twice.  Then I realize it needed to be three times.